zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -966,6 +966,117 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
   };
 
   var setTable = function(data, refresh) {
+    function addButtonMenuEvent(button, menu) {
```

Comment:
I am not sure about this.
The feature is closely related to the HandsonTable that's why I included it inside the setTable function.
It seems that the logic is hard to de-coupled from the table itself.
Anyway, I can try to de-couple if you insist.